astropy/modeling/functional_models.py

```diff
@@ -542,13 +542,7 @@ def inverse(self):
     @staticmethod
     def evaluate(x, factor):
         """One dimensional Scale model function"""
-        if isinstance(factor, u.Quantity):
-            return_unit = factor.unit
-            factor = factor.value
-        if isinstance(x, u.Quantity):
-            return (x.value * factor) * return_unit
-        else:
-            return factor * x
+        return factor * x
```

Comment:
I think `Shift` should be changed in the same way.

Reply:
Looking at this more closely, this change should be reversed. With this change using inputs with units does not work correctly. For example,
```
scl = Scale(2 * u.m)
scl(1 * u.m)
<Quantity 2.0 m2>
```

Reply:
having just removed this, tried to fix it and given up I think this is the right behavor. At least I can't think of another implementation that does "the right thing".

Reply:
@Cadair Do you agree that the above example is incorrect? If so I guess one could never have the model initialized with units and pass inputs with units at the same time. 
Or do you think scaling shoul dbe able to change units in the way it does above?

Reply:
I mean I'm OK doing it this way, we just have to make sure it's explained somewhere.